staging/mausezahn.c

```diff
@@ -671,6 +671,15 @@ int getopts (int argc, char *argv[])
 				tx.ip_src_rand = 1;
 				tx.ip_src_h  = (u_int32_t) ( ((float) rand()/RAND_MAX)*0xE0000000); //this is 224.0.0.0
 			}
+			else if (ipv6_mode && (strstr(tx.ip_src_txt, ".") || strstr(tx.ip_dst_txt, "."))) {
+			       fprintf(stderr, "You are using IPv6 mode but are providing IPv4 src/dst addresses.\n");
+			       return 1;
+			}
+			else if (!ipv6_mode && (strstr(tx.ip_src_txt, ":") || strstr(tx.ip_dst_txt, ":"))) {
+			       fprintf(stderr, "You are using IPv4 mode (the default) but are providing IPv6 src/dst addresses.\n");
+			       fprintf(stderr, "Please use the -6 option for IPv6 mode.\n");
+			       return 1;
+			}
```

Comment:
1. ip_dst_txt checking should be moved to the section where -B option is checking:
705                 // Set destination IP address:
 706                 if (strlen(tx.ip_dst_txt)) {  // option -B has been specified
                           ...
                           {somewhere here}
                           ...
  
2. It is better to explicitly parse ip_src_txt & ip_dst_txt addreses to check theirs AF_XXX family number because
the second case will fail:
    sudo mausezahn/mausezahn lo -6 -A evil.com -c 1 -t tcp "flags=syn"